mcm/tests/campaign/test_campaign.py

```diff
@@ -0,0 +1,268 @@
+        return self.mcm._get(endpoint)
+
+
+class TestCampaign(EntityTest):
+    """
+    Test the campaign API.
+    """
+
+    @property
+    def entity_api(self) -> CampaignAPI:
+        mcm: McM = McM(config=config, role=Roles.ADMINISTRATOR)
+        return CampaignAPI(mcm)
+
+    def test_create(self):
+        # Invalid name
+        invalid_mockup = self.entity_api.mockup()
+        invalid_mockup["prepid"] = "Run3Campaign#1"
+        content, response = self.entity_api.create(invalid_mockup)
+
+        assert response.status_code == 200
+        assert content["results"] == False
+
+        # Valid campaign
+        valid_mockup = self.entity_api.mockup()
+        content, response = self.entity_api.create(valid_mockup)
+
+        assert response.status_code == 200
+        assert content["results"] == True
+
+    def test_retrieve(self):
+        # The campaign exists
+        mockup = self.entity_api.mockup()
+        self.entity_api.create(mockup)
+        retrieved = self.entity_api.retrieve(mockup)
+        del mockup["history"]
+        del retrieved["history"]
+        del retrieved["_rev"]
+
+        assert mockup == retrieved
+
+        # The campaign doesn't exists
+        not_exists = self.entity_api.retrieve(self.entity_api.mockup())
+        assert not_exists == None
+
+    def test_update(self):
+        # The campaign is updated to a valid status
+        example = self.entity_api.example()
+        example["cmssw_release"] = "CMSSW_14_0_0"
+        content, response = self.entity_api.update(example)
+        retrieved = self.entity_api.retrieve(example)
+
+        assert response.status_code == 200
+        assert content["results"] == True
+        assert example["_rev"] != retrieved["_rev"]
+        assert retrieved["cmssw_release"] == "CMSSW_14_0_0"
+
+        # Update a campaign that doesn't exist.
+        example = self.entity_api.example()
+        example["prepid"] = "Run3Campaign#2"
+        content, response = self.entity_api.update(example)
+
+        assert response.status_code == 200
+        assert content["results"] == False
+
+    @pytest.mark.skip("Takes too much time to compute")
+    def test_delete(self):
+        pass
+
+    def test_cmssw_drivers(self):
+        example = self.entity_api.example()
+        content, response = self.entity_api.cms_drivers(example)
+        driver: str = content["results"][0]["default"]
+        event_content = example["sequences"][0]["default"]["eventcontent"]
+        event_content_param = ",".join(event_content)
+
+        assert response.status_code == 200
+        assert driver.startswith("cmsDriver.py")
+        assert event_content_param in driver
+
+        # The campaign doesn't exists
+        mockup = self.entity_api.mockup()
+        content, response = self.entity_api.cms_drivers(mockup)
+        assert response.status_code == 200
+        assert content["results"] == False
+
+    @pytest.mark.skip("Takes too much time to computes and renders a html page")
```

Comment:
What makes it so slow in a mock database with just a few records? I think it should be implemented but marked as [`slow`](https://docs.pytest.org/en/7.1.x/how-to/mark.html#mark). For HTML we could test `Content-Type` (if it's set correctly...) or use `html.parser`